module/x/peggy/abci.go

```diff
@@ -24,13 +24,8 @@ func slashing(ctx sdk.Context, k keeper.Keeper) {
 	ValsetSlashing(ctx, k, params)
 	BatchSlashing(ctx, k, params)
 	// TODO slashing for arbitrary logic is missing
-	ClaimsSlashing(ctx, k, params)
-
-	// #4 condition (stretch goal)
-	// TODO: lost eth key or delegate key
-	// 1. submit a message signed by the priv key to the chain and it slashes the validator who delegated to that key
-	// return
 
+	// TODO: prune validator sets, older than 6 months, this time is chosen out of an abundance of caution
```

Comment:
What is the reason for keeping validator sets around outside the unbonding period?

Still catching up on the design nuances, I may have missed something.

Reply:
The unbonding period *should* be a safe amount of time to keep validator sets around. But why not be more conservative and use a larger multiple?